src/tracks/InstrumentTrack.cpp

```diff
@@ -970,6 +987,87 @@ void InstrumentTrack::loadTrackSpecificSettings( const QDomElement & thisElement
 	unlock();
 }
 
+bool InstrumentTrack::processPresetSelectEvents(const MidiEvent& event)
+{
+	if (m_midiPort.m_captureProgramChangeModel.value() == 0)
+	{
+		return false;
+	}
+
+	auto policy = static_cast<MidiPort::PresetSelectPolicy>(
+			m_midiPort.m_presetSelectPolicyModel.value());
+
+	if (event.type() == MidiProgramChange)
+	{
+		m_programNumber = event.key();
+
+		if (policy == MidiPort::BankSelectIgnore)
+		{
+			m_programBankMSB = BANK_NONE;
+			m_programBankLSB = BANK_NONE;
+		}
+		return true;
+	}
+
+	else if (event.type() == MidiControlChange &&
+			event.controllerNumber() == MidiControllerBankSelectMSB)
+	{
+		switch (policy)
+		{
+		case MidiPort::BankSelectIgnore:
+			return false;
+		case MidiPort::BankSelectMSB:
+			/* Only MSB is meaningful and is the only bank select
+			 * message, so it should be actually treated as LSB.
+			 * MSB should be set to zero. */
+			m_programBankMSB = 0;
+			m_programBankLSB = event.controllerValue();
+			return true;
+		case MidiPort::BankSelectBoth:
+			/* Both MSB and LSB are meaningful, so don't touch
+			 * LSB here unless it is set to None (-1) - in this case reset it to 0. */
+			m_programBankMSB = event.controllerValue();
+			if (m_programBankLSB == BANK_NONE) {
+				m_programBankLSB = 0;
+			}
+			return true;
+		}
+	}
+
+	else if (event.type() == MidiControlChange &&
+			event.controllerNumber() == MidiControllerBankSelectLSB)
+	{
+		switch (policy)
+		{
+		case MidiPort::BankSelectIgnore:
+		case MidiPort::BankSelectMSB:
+			return false;
+		case MidiPort::BankSelectBoth:
+			m_programBankLSB = event.controllerValue();
+			if (m_programBankMSB == BANK_NONE) {
+				m_programBankMSB = 0;
+			}
+			return true;
+		}
+	}
+
+	return false;
+}
+
+
+void InstrumentTrack::changePreset()
+{
+	if (m_instrument)
```

Comment:
Can we invert the guard to return early and not double indent the whole function?

Reply:
done